README.md

```diff
@@ -1,93 +1,165 @@
+  * When complete the HEC token should look something like this:
+
+  ![hec_token](https://raw.githubusercontent.com/puppetlabs/puppetlabs-splunk_hec/v0.8.1/docs/images/hec_token.png)
+
+3. Install the module and add the `splunk_hec` class to the **PE Master** node group.
+  * Install the `splunk_hec` module on your Puppet Primary Server.
+      * `puppet module install puppetlabs-splunk_hec`
+  * In the PE console, navigate to `Node groups` and expand `PE Infrastructure`.
+  * Select `PE Master` and navigate to the `Classes` tab.
+  * Add the `splunk_hec` class.
+  * Configure the following parameters:
+
+      ```
+      enable_reports = true
+      manage_routes = true
+      splunk_token = <TOKEN GENERATED IN STEP 2>
+      url = something like https://splunk-8.splunk.internal:8088/services/collector
+      include_api_collection = true
+      ```
+      
+  * Commit the changes.
+  * Run Puppet on the node group; this will cause a restart of the `pe-puppetserver` service.
+
+4. Log into the Splunk console and search `index=* sourcetype=puppet:summary`, if everything was done properly you should see the reports (and soon facts) from the systems in your Puppet environment.
+
+## Custom Installation
+
+> **Please Note**: If you are installing this module using a `control-repo`, you must have `splunk_hec` in your production environment's `Puppetfile` so the Puppet Server process can properly load the required libraries. You can then create a feature branch to enable them and test the configuration, but the libraries **must be** in `production`; otherwise the feature branch won't work as expected. If your Puppet Server is in a different environment, please add this module to the `Puppetfile` in that environment as well.
+
+The steps below will help install and troubleshoot the report processor on a standard Puppet Primary Server; including manual steps to configure compilers (Puppet Servers), and to use the included `splunk_hec` class. Because one is modifying production machines, these steps allow you to validate your settings before deploying the changes live. 
+
+1. Install the Puppet Report Viewer app in Splunk. This will import the needed source types to configure Splunk's HTTP Endpoint Collector (HEC) and provide a dashboard that will show the reports once they are sent to Splunk.
+
+2. Create a Splunk HEC Token or use an existing one that sends to `main` index and **does not** have acknowledgement enabled. Follow the steps provided by Splunk's [Getting Data In Guide](http://docs.splunk.com/Documentation/Splunk/latest/Data/UsetheHTTPEventCollector) if you are new to HTTP Endpoint Collectors.
+
+3. Install this Puppet module in the environment that your Puppet Servers are using (e.g. `production`).
+
+4. Run `puppet plugin download` on your Puppet Servers to sync the content. Some users with strict permissions may need to run `umask 022` first.
+
+  * **Please Note**: If permissions are too restrictive you may encounter the following error:
```

Comment:
Can we add "in the Puppet Server logs" to this?

Reply:
Just added that bit to that line.